### PR TITLE
Start creating docs for nbformat

### PR DESCRIPTION
**Pull request by takluyver**
The format description has been copied in from ipython/ipython, the API docs are new.

### CODE REVIEW

File: docs/format_description.rst [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+    {
+      "output_type" : "stream",
+      "name" : "stdout", # or stderr
+      "text" : ["multiline stream text"],
+    }
+
+.. versionchanged:: nbformat 4.0
+
+    The keys ``stream`` key was changed to ``name`` to match
```
Comment (damianavila) [inline docs/format_description.rst]: `The keys`... and the `key` again...

File: docs/format_description.rst [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+Notebook metadata
+-----------------
+
+The following metadata keys are defined at the notebook level:
+
+=========== =============== ==============
+Key         Value           Interpretation
+=========== =============== ==============
+kernelspec  dict            A :ref:`kernel specification <kernelspecs>`
+signature   str             A hashed :ref:`signature <notebook_security>` of the notebook
```
Comment (damianavila) [inline docs/format_description.rst]: The signature is no longer stored in the ipynb, IIRC...

**Comment by damianavila** [comment]
Minor comments, otherwise, LGTM

**Comment by takluyver** [comment]
Thanks @damianavila , I've pushed a fix for those points.

**Comment by damianavila** [comment]
LGTM now... thanks @takluyver

**Comment by Carreau** [comment]
Merging. Let's iterate on that.

**Comment by takluyver** [comment]
Docs are now built at http://nbformat.readthedocs.org/en/latest/index.html

**Pull request merged**